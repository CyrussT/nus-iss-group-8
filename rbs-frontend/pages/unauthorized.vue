<script lang="ts" setup>
</script>

<template>
  <div class="flex flex-col items-center justify-center min-h-screen">
    <div class="p-8 rounded-lg shadow-md">
      <h1 class="text-3xl font-bold mb-4 ml-4">Unauthorized Access</h1>
      <p class="mb-6">You do not have permission to access this page.</p>
      <div class="flex justify-center">
        <NuxtLink 
          to="/login" 
          class="bg-blue-500 hover:bg-blue-600 text-white font-semibold py-2 px-4 rounded"
        >
          Back to Login
        </NuxtLink>
      </div>
    </div>
  </div>
</template>
