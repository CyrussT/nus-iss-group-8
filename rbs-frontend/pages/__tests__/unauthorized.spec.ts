import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import unauthorized from '../unauthorized.vue';

describe('unauthorized', () => {
  it('renders correctly', () => {
    const wrapper = mount(unauthorized, {
      global: {
        stubs: {
          UButton: true,
          UIcon: true
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
