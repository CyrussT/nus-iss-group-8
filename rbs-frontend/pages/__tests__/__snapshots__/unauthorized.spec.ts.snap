// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unauthorized > renders correctly 1`] = `
"<div class="flex flex-col items-center justify-center min-h-screen">
  <div class="p-8 rounded-lg shadow-md">
    <h1 class="text-3xl font-bold mb-4 ml-4">Unauthorized Access</h1>
    <p class="mb-6">You do not have permission to access this page.</p>
    <div class="flex justify-center"><a href="/login" class="bg-blue-500 hover:bg-blue-600 text-white font-semibold py-2 px-4 rounded"> Back to Login </a></div>
  </div>
</div>"
`;
